### PR TITLE
Migration compatibility for WorkloadBuilder service

### PR DESCRIPTION
**Pull request by venu-sivanadham**
Workload builder service can migrate(lift & shift) Moodle application running on on-premises VMWare infrastructure to Azure.
Workload builder service uses ARM templates from this Moodle repo to prepare the infra in Azure. Below changes are made in this repo to support migration flow:
1. Added migration specific entry-point/wrapper template which will invoke main template with appropriate parameters.
2. Added another wrapper template to create file-share storage account. In migration flow, the Moodle code/config/data will be migration from on-prem to azure file-share as 1st step before creating complete infrastructure.
3. Added few parameters to azuredeploy template to support migration flow.
4. Changes to install_moodle.sh to skip fresh Moodle installation commands. And also added couple of helper functions to update Moodle config settings in the migrated config file.

### CODE REVIEW

File: migration/azure-fileshare-sa-deploy.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                "Standard_ZRS",
+                "Premium_LRS"
+            ],
+            "metadata": {
+                "description": "Storage Account type."
+            },
+            "type": "string"
+        },
+        "fileServerDiskSize": {
+            "defaultValue": "100",
```
Comment (UmakanthOS) [inline migration/azure-fileshare-sa-deploy.json:19]: Wondering if we should default to 1024GB for max IOPS across the board. Thoughts @naioja ?

File: nested/storageAccount.json [hunk 2]
```diff
@@ -93,4 +93,4 @@
             "type": "string"
         }
     }
-}
```
Comment (UmakanthOS) [inline nested/storageAccount.json:96]: No change to this file yea?

**Review by UmakanthOS — APPROVED** [review]
LGTM. Left a couple of comments but no need to address them.

File: migration/azure-fileshare-sa-deploy.json (hunk 1 above)
Comment (venu-sivanadham) [reply migration/azure-fileshare-sa-deploy.json]: make sense, and changed it to 1024 GB as default file server size.

File: nested/storageAccount.json (hunk 2 above)
Comment (venu-sivanadham) [reply nested/storageAccount.json]: you are right, no intended change in this file.
Fixed it.

**Pull request merged**